packages/liferay-jest-junit-reporter/index.js

```diff
@@ -2,41 +2,95 @@
 
 const xml = require('xml');
 const fs = require('fs');
+const path = require('path');
+const stripAnsi = require('strip-ansi');
 
-module.exports = (report) => {
-  const generalMetrics = {
-    _attr: {
-      hostname: '',
-      id: 0,
-      errors: 0,
-      failures: report.numFailedTests,
-      name: 'Jest',
-      package: '',
-      skipped: 0,
-      tests: report.numTotalTests,
-      time: 0,
-      timestamp: report.startTime
-    }
-  };
-
-  let testResults = report.testResults
-    .reduce((results, suite) => suite.testResults.length ? results.concat(suite.testResults) : results, [])
-    .map(testCase => {
-      return {
-        'testcase': [{
-          _attr: {
-            classname: testCase.ancestorTitles[0],
-            name: `${testCase.ancestorTitles.join(' ')} ${testCase.title}`,
-            time: testCase.duration / 1000
-          }
-        }]
-      };
-    });
-
-  fs.writeFileSync('TEST-frontend-js.xml', xml(
-    {'testsuite': [generalMetrics, ...testResults]},
-    { declaration: true, indent: '  '}
-  ));
-
-  return report;
+const NEW_LINE = '\n';
+
+const RELATIVE_APPS_DIR = '/modules/apps/';
+
+const APPS_DIR = new RegExp('^.+' + RELATIVE_APPS_DIR);
+
+function formatDirectoryPath(dirPath) {
+	return dirPath.replace(APPS_DIR, '').replace(/\//g, '.');
+}
+
+module.exports = report => {
+	const generalMetrics = {
+		_attr: {
+			hostname: '',
+			id: 0,
+			errors: 0,
+			failures: report.numFailedTests,
+			name: 'Jest',
```

Comment:
I know this was here way before this, so I'm just pointing it out for the future. Having a hardcoded `Jest` name here will group all tests under a `Jest` hierarchy in the Jenkins reports. See attached image:

<img width="1114" alt="screen shot 2019-01-22 at 10 13 06" src="https://user-images.githubusercontent.com/905006/51524505-83330a80-1e2e-11e9-88cb-b1ba79d34a77.png">

@john-co, what do you think about this? Wouldn't it be better if we could avoid this additional level and just show the module there, for example?

Reply:
@jbalsas, hardcoding the name there does not control the package name. This will be fixed with the added commits.

**For detailed explanation on modifying package name:**
The package name itself is controlled by trimming off what comes after the last `.` in the classname value in each testcase. In this case Management Toolbar TEST-frontend-js.xml had `Jest.ManagementToolbar` as classname value so jenkins takes Jest as the package name and uses ManagementToolbar as the child in the navigation structure. If the classname value does not have a `.` then it defaults to `root` as package name.

**To see an example of what this updated jest-reporter will produce:**
Using the generated TEST*.xml file https://gist.github.com/bryceosterhaus/c15cf831821dd9b7240b911bcf9eda45, it will produce the following results on PR tester:
https://github.com/john-co/liferay-portal/pull/49#issuecomment-456499485
And on jenkins:
https://test-1-22.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/135603/AXIS_VARIABLE=0,label_exp=!master/testReport/

![selection_058](https://user-images.githubusercontent.com/35235266/51556739-da85aa80-1e2f-11e9-9e02-fb9deee6e4f4.png)

Let me know if this sample output will be helpful or if you'd like to see other modifications.

**Note:** I included a manually modified TEST-frontend-js.xml file for managmeent toolbar. This is ideally what I'd like the PR tester output to show.

Reply:
Sounds good, John, thanks for the explanation!

Based on the screenshot... couldn’t we do something so the testnames are a bit more readable?

Reply:
Maybe we could shorten the packagename if that’s expected to be there... I don’t remember such long testNames or packageNames in the Java unit ones, though granted, I don’t pay that much attention to them 😅

Reply:
In the java tests that I looked at, they actually are that long! In that screen shot the Test Name includes the packageName. But if you navigate through the links for package and then click the classname, you'll find the test names trimmed down.

Here's the view if you click the package `dynamic-data-mapping.dynamic-data-mapping-form-field-type.test.metal.js`
![selection_059](https://user-images.githubusercontent.com/35235266/51558783-2d159580-1e35-11e9-903a-55b5511e80c1.png)

And then if you click class `KeyValue`
![selection_060](https://user-images.githubusercontent.com/35235266/51558897-6ea64080-1e35-11e9-875f-0a7e953ebe0a.png)

Reply:
It's only for the `For All Failed Tests` summary section is where it lists out the test's full path.

Reply:
If any further consolation is needed, the PR tester links will only show the class and test name.

![selection_063](https://user-images.githubusercontent.com/35235266/51562891-be8a0500-1e3f-11e9-98e2-db7c3a65fcf9.png)

Reply:
Awesome! Thanks for the details! 👍